core/src/main/java/com/es/core/cart/HttpSessionCartService.java

```diff
@@ -43,7 +43,7 @@ public void addPhone(Long phoneId, Long quantity) {
 
     @Override
     public void update(Map<Long, Long> items) {
-        Set<CartItem> cartItemsToBeSet = new HashSet<>(items.size());
+        Set<CartItem> cartItemsToBeSet = new LinkedHashSet<>(items.size());
```

Comment:
Please use Stream#collect

Reply:
\+ Not in scope of this task, but rather a previous one:
there is no point in retrieving the product from DAO in the #remove method of this class

Reply:
Fixed